src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -111,7 +111,27 @@ std::shared_ptr<MulticlassLabels> CARTree::apply_multiclass(std::shared_ptr<Feat
 	auto current=get_root()->as<bnode_t>();
 
 	require(current, "Tree machine not yet trained.");
-	return apply_from_current_node(data->as<DenseFeatures<float64_t>>(), current)->as<MulticlassLabels>();
+	if (auto subfeat_data =
+	        std::dynamic_pointer_cast<DenseSubSamplesFeatures<float64_t>>(data))
+	{
+		auto num_feat = subfeat_data->get_dim_feature_space();
+		auto num_vec = subfeat_data->get_num_vectors();
+		SGMatrix<float64_t> feature_matrix(num_feat, num_vec);
+		for (size_t i = 0; i < num_vec; i++)
+		{
+			SGVector<float64_t> v =
+			    subfeat_data->get_computed_dot_feature_vector(i);
+			linalg::add_col_vec(feature_matrix, i, v, feature_matrix, 0, 1);
```

Comment:
wouldn't it be better to use `sg_memcpy`? I am not sure if the compiler can optimise away the add in this situation and use memcpy instead

Reply:
I agree. But even better would be a method that hides this stuff under the hood imo

Reply:
you mean add something to linalg ?

Reply:
I talked to @gf712 who reminded me that there is an column iterator for SGMatrix. So you can use that together with memset. No need to touch linalg then